Building nightly with multiple Julia GC threads causes segfault

building Julia today (but not yesterday) I'm seeing a segfault. My guess is https://github.com/JuliaLang/julia/pull/48969 is to blame.
```
    LINK usr/lib/libjulia-codegen.so
    JULIA usr/lib/julia/corecompiler.ji

[72883] signal (11.1): Segmentation fault
in expression starting at iterators.jl:1004
ws_queue_push at /home/oscardssmith/Documents/Code/julia/src/work-stealing-queue.h:45 [inlined]
gc_ptr_queue_push at /home/oscardssmith/Documents/Code/julia/src/gc.c:1729 [inlined]
gc_try_claim_and_push at /home/oscardssmith/Documents/Code/julia/src/gc.c:1809
gc_queue_thread_local at /home/oscardssmith/Documents/Code/julia/src/gc.c:2850 [inlined]
_jl_gc_collect at /home/oscardssmith/Documents/Code/julia/src/gc.c:3102
ijl_gc_collect at /home/oscardssmith/Documents/Code/julia/src/gc.c:3436
maybe_collect at /home/oscardssmith/Documents/Code/julia/src/gc.c:913 [inlined]
jl_gc_pool_alloc_inner at /home/oscardssmith/Documents/Code/julia/src/gc.c:1263
jl_gc_pool_alloc_noinline at /home/oscardssmith/Documents/Code/julia/src/gc.c:1332 [inlined]
jl_gc_alloc_ at /home/oscardssmith/Documents/Code/julia/src/julia_internal.h:466 [inlined]
jl_gc_alloc at /home/oscardssmith/Documents/Code/julia/src/gc.c:3484
_new_array_ at /home/oscardssmith/Documents/Code/julia/src/array.c:134 [inlined]
_new_array at /home/oscardssmith/Documents/Code/julia/src/array.c:198 [inlined]
ijl_alloc_array_1d at /home/oscardssmith/Documents/Code/julia/src/array.c:436
jl_exprn at /home/oscardssmith/Documents/Code/julia/src/builtins.c:1416
scm_to_julia_ at /home/oscardssmith/Documents/Code/julia/src/ast.c:600
scm_to_julia_ at /home/oscardssmith/Documents/Code/julia/src/ast.c:604
scm_to_julia_ at /home/oscardssmith/Documents/Code/julia/src/ast.c:604
scm_to_julia_ at /home/oscardssmith/Documents/Code/julia/src/ast.c:604
scm_to_julia at /home/oscardssmith/Documents/Code/julia/src/ast.c:435
jl_call_scm_on_ast_and_loc at /home/oscardssmith/Documents/Code/julia/src/ast.c:862
ijl_expand_stmt_with_loc at /home/oscardssmith/Documents/Code/julia/src/ast.c:1315
jl_eval_module_expr at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:213 [inlined]
jl_toplevel_eval_flex at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:736
jl_parse_eval_all at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:1068
ijl_load_ at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:1115
unknown function (ip: 0x7f99f81696b9)
unknown function (ip: 0x7f99f81695dd)
jl_apply at /home/oscardssmith/Documents/Code/julia/src/julia.h:1963 [inlined]
do_call at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:125
eval_value at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:222
eval_stmt_value at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:173 [inlined]
eval_body at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:620
jl_interpret_toplevel_thunk at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:760
jl_toplevel_eval_flex at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:934
jl_eval_module_expr at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:215 [inlined]
jl_toplevel_eval_flex at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:736
ijl_toplevel_eval_in at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:993
unknown function (ip: 0x7f99f8168a19)
jl_apply at /home/oscardssmith/Documents/Code/julia/src/julia.h:1963 [inlined]
do_call at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:125
eval_value at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:222
eval_stmt_value at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:173 [inlined]
eval_body at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:620
jl_interpret_toplevel_thunk at /home/oscardssmith/Documents/Code/julia/src/interpreter.c:760
jl_toplevel_eval_flex at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:934
jl_parse_eval_all at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:1068
ijl_load_ at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:1115
ijl_load at /home/oscardssmith/Documents/Code/julia/src/toplevel.c:1128
exec_program at /home/oscardssmith/Documents/Code/julia/src/jlapi.c:541
true_main at /home/oscardssmith/Documents/Code/julia/src/jlapi.c:594
jl_repl_entrypoint at /home/oscardssmith/Documents/Code/julia/src/jlapi.c:731
main at /home/oscardssmith/Documents/Code/julia/cli/loader_exe.c:58
unknown function (ip: 0x7f9a046f2d8f)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /home/oscardssmith/Documents/Code/julia/usr/bin/julia (unknown line)
Allocations: 841816 (Pool: 841733; Big: 83); GC: 0
Segmentation fault (core dumped)
make[1]: *** [sysimage.mk:61: /home/oscardssmith/Documents/Code/julia/usr/lib/julia/corecompiler.ji] Error 139
make: *** [Makefile:97: julia-sysimg-ji] Error 2
```

--- a/src/init.c
+++ b/src/init.c
@@ -872,6 +872,8 @@ static NOINLINE void _finish_julia_init(JL_IMAGE_SEARCH rel, jl_ptls_t ptls, jl_
     if (jl_base_module == NULL) {
         // nthreads > 1 requires code in Base
         jl_atomic_store_relaxed(&jl_n_threads, 1);
+        jl_n_markthreads = 0;
+        jl_n_sweepthreads = 0;
         jl_n_gcthreads = 0;
     }
     jl_start_threads();

--- a/src/julia.h
+++ b/src/julia.h
@@ -1766,6 +1766,8 @@ JL_DLLIMPORT jl_value_t *jl_get_libllvm(void) JL_NOTSAFEPOINT;
 extern JL_DLLIMPORT int jl_n_threadpools;
 extern JL_DLLIMPORT _Atomic(int) jl_n_threads;
 extern JL_DLLIMPORT int jl_n_gcthreads;
+extern int jl_n_markthreads;
+extern int jl_n_sweepthreads;
 extern JL_DLLIMPORT int *jl_n_threads_per_pool;
 
 // environment entries

